src/ovirtapi/transform.js

```diff
@@ -708,10 +708,11 @@ const Cluster = {
 //
 const Nic = {
   toInternal ({ nic }: { nic: ApiNicType }): NicType {
+    const { mac: { address: nicMacAddress = '' } = {} } = nic
     const ips =
       nic.reported_devices && nic.reported_devices.reported_device
         ? nic.reported_devices.reported_device
-          .filter(device => !!device.ips && !!device.ips.ip)
+          .filter(device => !!device.ips && !!device.ips.ip && device.mac && device.mac.address === nicMacAddress)
```

Comment:
@sjd78  @sgratch 
Please comment on Mac filtering. [My opinion](https://github.com/oVirt/ovirt-web-ui/pull/1382#discussion_r607664825) is that we should try to "correct" backend - just present what we receive.

Reply:
I don't see any need to evaluate the reported_device's mac address unless multiple devices are returned. I tested a few running VMs and only ever had 1 returned.  The backend probably matches things up for us.

I wonder how a multiport passthrough nic would be reported.  That would be quite a rare occurrence.  Probably 1 nic for each port so it wouldn't be a problem anyway.

Reply:
@sjd78
Then we should report a bug and in the meantime just remove duplicates in the view (which does make sense since we show all the addresses together).

Reply:
> @sjd78
> Then we should report a bug and in the meantime just remove duplicates in the view (which does make sense since we show all the addresses together).

So filter it in the view instead of in the transform?  I'd rather have the transform just make the data nice and neat for the view.  A BZ on the rest side can be opened, but probably wouldn't be handled in time for us to use so we'd still need to filter anyway.

Reply:
@sjd78 @bamsalem 
>  I'd rather have the transform just make the data nice and neat for the view. 

As far as I understand we use the same data to display NICs details. IMHO NICs may have the same IPs and MACs in certain scenarios by design (migration, failover) or because of the delay(hardware device vs virtual device) or simply due to user error (incorrect configuration). The widget displaying the IP address summary should be prepared to handle that.

> A BZ on the rest side can be opened, but probably wouldn't be handled in time for us to use so we'd still need to filter anyway.

+1

Reply:
> @sjd78 @bamsalem
> 
> > I'd rather have the transform just make the data nice and neat for the view.
> 
> As far as I understand we use the same data to display NICs details. IMHO NICs may have the same IPs and MACs in certain scenarios by design (migration, failover) or because of the delay(hardware device vs virtual device) or simply due to user error (incorrect configuration). The widget displaying the IP address summary should be prepared to handle that.
> 

:facepalm: 

Ok so it will all depend on how the nic given to the VM (whose mac address is fixed in the VM definition) is tied to the guest agent's MAC and IP address reporting.  To make it even more crazy, just think of a VM setup where (1) the nic's mac address is cloned so the reported mac won't match the nic's mac, or (2) if the system is setup to have multiple macvlan adpaters on top of the VM's nic, or (3) virtual nics bleeding through via the reported devices but don't really have a parent device (e.g. a tap/tun device).  If there is a macvlan setup, each new macvlan derived nic has it's own mac address.  There can basically be as many as you want, but they're all "children" of the base nic....

SO.....

Makes sense to do a few things:
 1. Don't filter out the "extra" devices `reported_devices` in the transform
 2. Just show all of the info on the Nics Card for each Nic (so no change)
 3. On the Details Card, only show the unique set of IP4/IP6 addresses (so add IPs to a set instead of a list)
 4. Open a BZ so the infra/guest agent guys can make sure the reporting is being done properly (until that is "fixed" the Nics card will show extra info per nic)

@rszwajko - I think number 3 is what you were getting at originally.

> > A BZ on the rest side can be opened, but probably wouldn't be handled in time for us to use so we'd still need to filter anyway.
> 
> +1

Reply:
I think current behaviour with mac filtering is ok since vmGuestAgentInterface.getMacAddress should fit nic.getMacAddress based on this backend code:
https://github.com/oVirt/ovirt-engine/blob/99bfdd6d1f3bf27c930260aa08921c6da99142e8/backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/resource/BackendNicHelper.java#L41

In addition, the resulted list of ips fit the list shown on webadmin.
There is a bug on rest so we can open BZ for the rest query but the filtering seems ok anyway to me.

Reply:
As long as all we want to do it show any IP addresses reported by the guest agent explicitly with the Nic's mac address, sure.  Doing the `Set` work I suggested in `DetailsCard` will remove any duplicate IPs found.

AFAICT, the REST call should then make sure only the proper guest agent reported devices are attached to the actual nic.  The `BackendNicHelper` looks like it should do that, but the real calls don't seem to be working that way.